dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -0,0 +1,206 @@
+  private val PURPOSE = 585
+
+  private val rValueAccount: HDAccount = {
+    val coin = HDCoin(HDPurpose(PURPOSE), HDCoinType.fromNetwork(conf.network))
+    HDAccount(coin, 0)
+  }
+
+  private val chainIndex = 0
+
+  private val signingKey: ECPrivateKey = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+    // Use a different chain index so we don't reuse a key
+    val chain = chainIndex + 1
+
+    val path = BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chain'/0'")
+
+    // what was previously used so we other wallets could find funds
+//    extPrivateKey
+//      .deriveChildPrivKey(SegWitHDPath(signingKeyHDAddress))
+//      .key
+
+    extPrivateKey.deriveChildPrivKey(path).key
+  }
+
+  val publicKey: SchnorrPublicKey = signingKey.schnorrPublicKey
+
+  def stakingAddress(network: BitcoinNetwork): Bech32Address =
+    Bech32Address(P2WPKHWitnessSPKV0(signingKey.publicKey), network)
+
+  protected[bitcoins] val rValueDAO: RValueDAO = RValueDAO()
+  protected[bitcoins] val eventDAO: EventDAO = EventDAO()
+  protected[bitcoins] val eventOutcomeDAO: EventOutcomeDAO = EventOutcomeDAO()
+
+  private def getPath(keyIndex: Int): BIP32Path = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+
+    BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chainIndex'/$keyIndex'")
+  }
+
+  private def getKValue(rValDb: RValueDb): ECPrivateKey =
+    getKValue(rValDb.label, rValDb.path)
+
+  private def getKValue(label: String, path: BIP32Path): ECPrivateKey = {
+    require(path.forall(_.hardened),
+            s"Cannot use a BIP32Path with unhardened nodes, got $path")
+    val priv = extPrivateKey.deriveChildPrivKey(path).key
+    val hash =
+      CryptoUtil.sha256(
+        priv.schnorrNonce.bytes ++ CryptoUtil.serializeForHash(label))
+    val tweak = ECPrivateKey(hash.bytes)
+
+    priv.add(tweak)
+  }
+
+  def listEventDbs(): Future[Vector[EventDb]] = eventDAO.findAll()
+
+  def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
+
+  def listEvents(): Future[Vector[Event]] = {
+    for {
+      eventDbs <- eventDAO.findAll()
+      outcomes <- eventOutcomeDAO.findAll()
+    } yield {
+      val outcomesByNonce = outcomes.groupBy(_.nonce)
+      eventDbs.map(db => Event(db, outcomesByNonce(db.nonce)))
+    }
+  }
+
+  def createNewEvent(
+      label: String,
+      outcomes: Vector[String]): Future[EventDb] = {
+    for {
+      indexOpt <- rValueDAO.maxKeyIndex
+      index = indexOpt match {
+        case Some(value) => value + 1
+        case None        => 0
+      }
+
+      path = getPath(index)
+      nonce = getKValue(label, path).schnorrNonce
+
+      hash =
+        CryptoUtil.sha256(nonce.bytes ++ CryptoUtil.serializeForHash(label))
+      commitmentSig = signingKey.schnorrSign(hash.bytes)
```

Comment:
What exactly is `commitmentSig`?  I was under the assumption it was the final signature published, but it seems that it is just affirming that you actually own the private key? This seems like something that can be replaced if we put signatures on oracle addresses? https://github.com/discreetlogcontracts/dlcspecs/issues/99

Reply:
The `commitmentSig` is signing your R value to prove that the oracle is infact going to sign for this event / R value

Reply:
To be more clear they aren't signing the R value but instead `CryptoUtil.sha256(nonce.bytes ++ CryptoUtil.serializeForHash(label))`